python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/traffic-policies.j2

```diff
@@ -7,207 +7,200 @@
+         {{ protocol_cli }} destination port {{ protocol.dst_port }}
+{%                             endif %}
+{#                             -- source/destination field-set -- #}
+{%                             if protocol.src_field is arista.avd.defined and protocol.dst_field is arista.avd.defined and protocol.protocol | lower in ['tcp','udp'] %}
+         {{ protocol_cli }} source port field-set {{ protocol.src_field }} destination port field-set {{ protocol.dst_field }}
 {%                             elif protocol.src_field is arista.avd.defined and protocol.protocol | lower in ['tcp','udp'] %}
-{%                                 set protocol_cli = 'protocol ' ~ protocol.protocol %}
-{%                                 if protocol.flags is arista.avd.defined %}
-{%                                     set protocol_cli = protocol_cli ~ ' flags ' ~  protocol.flags | join(' ') %}
-{%                                 endif %}
-{%                                 set protocol_cli = protocol_cli ~ ' source port field-set ' ~ protocol.src_field %}
+         {{ protocol_cli }} source port field-set {{ protocol.src_field }}
+{%                             elif protocol.dst_field is arista.avd.defined and protocol.protocol | lower in ['tcp','udp'] %}
+         {{ protocol_cli }} destination port field-set {{ protocol.dst_field }}
+{%                             endif %}
+{%                             if protocol.icmp_type is arista.avd.defined and protocol.protocol | lower == 'icmp' %}
+         {{ protocol_cli }} type {{ protocol.icmp_type | arista.avd.natural_sort | join(' ') }} code all
+{%                             endif %}
+{%                             if protocol.protocol is arista.avd.defined and protocol.icmp_type is not arista.avd.defined and protocol.dst_field is not arista.avd.defined and protocol.src_field is not arista.avd.defined and protocol.dst_port is not arista.avd.defined and protocol.src_port is not arista.avd.defined and protocol.flags is not arista.avd.defined %}
          {{ protocol_cli }}
-{%                             elif protocol.icmp_type is arista.avd.defined(protocol.protocol) | lower == 'icmp' %}
-         protocol icmp type {{ protocol.icmp_type | join(' ') }} code all
-{%                             elif protocol.protocol | lower == 'neighbors' %}
-         protocol neighbors bgp
-{%                             elif protocol.protocol | lower != 'ip' %}
-         protocol {{ protocol.protocol }}
 {%                             endif %}
-{%                         endfor %}
-{%                     endif %}
-{#                     -- TTL MANAGEMENT -- #}
-{%                     if match.ttl is arista.avd.defined %}
+{%                         elif protocol.protocol is arista.avd.defined and protocol.protocol | lower == 'neighbors' and not flags.bgp and not flags.all %}
+         protocol neighbors bgp
+{%                             do flags.update({'neighbors': True}) %}
```

Comment:
```suggestion
{%                             break %}
```

Reply:
May be bgp and neighbors can be removed from flag which was set.

Reply:
updated